File: src/main/java/uk/co/threebugs/mochiwhattotrade3/setupgroups/SetupGroupsService.java
package uk.co.threebugs.mochiwhattotrade3.setupgroups;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@AllArgsConstructor
public class SetupGroupsService {

    private final SetupGroupsRepository setupGroupsRepository;
    private final SetupGroupsMapper setupGroupsMapper;

    public List<SetupGroupsDto> findAll() {
        return setupGroupsRepository.findAll()
                                    .stream()
                                    .map(setupGroupsMapper::toDto)
                                    .toList();
    }

}

File: src/main/kotlin/uk/co/threebugs/mochiwhattotrade3/setupgroups/SetupGroupsService.kt
package uk.co.threebugs.mochiwhattotrade3.setupgroups

import org.springframework.stereotype.Service

@Service
class SetupGroupsService (
     private val setupGroupsRepository: SetupGroupsRepository,
     private val setupGroupsMapper: SetupGroupsMapper
)
{
    fun findAll(): List<SetupGroupsDto> {
        return setupGroupsRepository.findAll()
                .stream()
                .map { setupGroups: SetupGroups -> setupGroupsMapper.toDto(setupGroups) }
                .toList()
    }
}
